pkg/controllers/secretstore/clustersecretstore_controller.go

```diff
@@ -71,12 +86,178 @@ func (r *ClusterStoreReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	})
 }
 
+// handleClusterSecretStoreFinalizer manages the finalizer for ClusterSecretStores
+// It adds a finalizer when there are PushSecrets with DeletionPolicy=Delete that reference this store
+// and removes it when there are no such PushSecrets
+func (r *ClusterStoreReconciler) handleClusterSecretStoreFinalizer(ctx context.Context, css *esapi.ClusterSecretStore, log logr.Logger) error {
+	// Check if this ClusterSecretStore is referenced by any PushSecrets with DeletionPolicy=Delete
+	hasPushSecretsWithDeletePolicy, err := r.hasPushSecretsWithDeletePolicy(ctx, css)
+	if err != nil {
+		return fmt.Errorf("failed to check PushSecrets: %w", err)
+	}
+
+	// If the store is being deleted and has the finalizer, check if we can remove it
+	if !css.ObjectMeta.DeletionTimestamp.IsZero() {
+		if controllerutil.ContainsFinalizer(css, secretStoreFinalizer) {
+			if hasPushSecretsWithDeletePolicy {
+				// Cannot remove finalizer yet, there are still PushSecrets that need this store
+				log.Info("cannot remove finalizer, there are still PushSecrets with DeletionPolicy=Delete that reference this store")
+				return nil
+			}
+			// Safe to remove finalizer
+			controllerutil.RemoveFinalizer(css, secretStoreFinalizer)
```

Comment:
nit: use the `updated` bool in RemoveFinalizer to trigger your Update instead of checking manually